examples/spa/associative_memory.ipynb

```diff
@@ -0,0 +1,462 @@
+      "The examples in this tutorial demonstrate how to use the AM module, store and recall patterns.\n",
+      "Advanced functionality of the module such as the recall of multiple memories similar to the input are also presented."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "# python libraries and plotting tools\n",
+      "import numpy as np\n",
+      "import matplotlib.pyplot as plt\n",
+      "%matplotlib inline\n",
+      "\n",
+      "import nengo\n",
+      "from nengo import spa\n",
+      "%load_ext nengo.ipynb"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Step 1: Creating an Associative Memory\n",
+      "\n",
+      "We start by creating a set of patterns the AM will store. \n",
+      "The vocabulary in this example contains five words: `ORANGE, APRICOT, CHERRY, STRAWBERRY` and `APPLE`.\n",
+      "Each word is represented as a semantic pointer, a $n$-dimensional vector.\n",
+      "When creating a vocabulary, we specify the number of dimensions for all semantic pointers.\n",
+      "Then, we add the words to a vocabulary with `parse`:"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": true,
+     "input": [
+      "dim = 32\n",
+      "vocab = spa.Vocabulary(dimensions=dim)\n",
+      "\n",
+      "words = ['ORANGE', 'APRICOT', 'CHERRY', 'STRAWBERRY', 'APPLE']\n",
+      "\n",
+      "for word in words:\n",
+      "    vocab.parse(word)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "Next, we create an autoassociative memory with a corresponding set of stored patterns.\n",
+      "The memory is created within a `nengo.Network` module."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "with nengo.Network('AssociativeMemory', seed=1) as model1:\n",
+      "    assoc_mem = spa.AssociativeMemory(input_vocab=vocab)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "After the network has been created, we can test its functionality by presenting a semantic pointer at the input and comparing it with a semantic pointer at the output after running a simulation. The comparison is done by calculating the dot product. If the similarity between the output vector and the input vector is close to one, we can say that the associative memory successfully retrieved the pattern.\n",
+      "To achieve this task, we need to introduce two additional objects: `nengo.Node` and `nengo.Probe`. The `nengo.Node` object provides input to the associative memory. \n",
```

Comment:
I actually don't recall any discussion about `spa.Input`...  but I will
look into that!
I just used `spa.Node`s as they were also used in the associative memory
tests...


On 13 November 2015 at 19:40, Jan Gosmann <notifications@github.com> wrote:

> In examples/spa/associative_memory.ipynb
> <https://github.com/nengo/nengo/pull/895#discussion_r44850048>:
>
> > +     "cell_type": "code",
> > +     "collapsed": false,
> > +     "input": [
> > +      "with nengo.Network('AssociativeMemory', seed=1) as model1:\n",
> > +      "    assoc_mem = spa.AssociativeMemory(input_vocab=vocab)"
> > +     ],
> > +     "language": "python",
> > +     "metadata": {},
> > +     "outputs": []
> > +    },
> > +    {
> > +     "cell_type": "markdown",
> > +     "metadata": {},
> > +     "source": [
> > +      "After the network has been created, we can test its functionality by presenting a semantic pointer at the input and comparing it with a semantic pointer at the output after running a simulation. The comparison is done by calculating the dot product. If the similarity between the output vector and the input vector is close to one, we can say that the associative memory successfully retrieved the pattern.\n",
> > +      "To achieve this task, we need to introduce two additional objects: `nengo.Node` and `nengo.Probe`. The `nengo.Node` object provides input to the associative memory. \n",
>
> I agree with @Seanny123 <https://github.com/Seanny123>. Also, is there a
> reason to not use spa.Input? I think, we talked about that shortly, but I
> forgot what you said. [image: :frowning:] As this example is using a SPA
> module, I would stick to the SPA methods of connecting things where
> possible.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nengo/nengo/pull/895/files#r44850048>.
>